scripts/Muon/GUI/Common/plotting_widget/plotting_widget_model.py

```diff
@@ -7,202 +7,238 @@
 from __future__ import (absolute_import, division, print_function)
 import numpy as np
 
-from mantidqt.plotting.functions import plot
 from mantid.api import AnalysisDataService
+import matplotlib.pyplot as plt
+
+legend_text_size = 7
 
 
 class PlotWidgetModel(object):
-    def __init__(self, dockable_plot_window=None):
-        """
-        :param plotting_window_model: This is the plotting manager class to use
-        """
-        self.plot_figure = dockable_plot_window
+
+    def __init__(self):
         self._plotted_workspaces = []
         self._plotted_workspaces_inverse_binning = {}
         self._plotted_fit_workspaces = []
-        self.plotted_group = ''
+        self.tiled_plot_positions = {}
+        self._number_of_axes = 1
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Properties
+    # ------------------------------------------------------------------------------------------------------------------
+
+    @property
+    def number_of_axes(self):
+        """
+        This property is needed as the view will generate an even number of axes, which can exceed the number of axes
+        required by the model
+        """
+        return self._number_of_axes
 
     @property
     def plotted_workspaces(self):
         """
-        This property is needed to check whether all the workspaces contained in the list are still on the graph. They can
-        be removed from the graph from the figure window without this class knowing.
-        :return:
+        A list of the plotted workspaces in the figure
         """
-        self._plotted_workspaces = [item for item in self._plotted_workspaces if item in
-                                    self.plot_figure.gca().tracked_workspaces.keys()]
         return self._plotted_workspaces
 
+    # NOTE: Not sure if this property is necessary, or used for anything
```

Comment:
Could we check and remove it if it is not required?

Reply:
The property is used somewhere, so I'm going to decide that this is still needed, and therefore i have removed the comment.